src/services/formatting/formatting.ts

```diff
@@ -511,6 +498,13 @@ namespace ts.formatting {
                         }
                     }
                 },
+                getCopyForSpecificChild(child) {
+                    if (!delta) {
+                        // delta value is already 0, so do not copy
+                        return this;

```

Comment:
we rarely use manipulations that involve `this` in compiler/language service. I'd prefer if we can lift members of object literal to functions and then just call functions